datajoint/base_relation.py

```diff
@@ -70,8 +71,14 @@ def declare(self):
         Loads the table heading. If the table is not declared, use self.definition to declare
         """
         if not self.is_declared:
-            self.connection.query(
-                declare(self.full_table_name, self.definition, self._context))
+            if self.definition is not None and self.definition is not Ellipsis:
+                self.connection.query(
+                    declare(self.full_table_name, self.definition, self._context))
+            else:
+                name = self.__class__.__name__
+                warn("""Table for %s is not declared and definition is not defined.
```

Comment:
If we go with throwing an error, then the `schema` can specifically catch and ignore the error (and may be throw a warning at that point).